bin/tls-include.sh

```diff
@@ -105,15 +105,22 @@ function create_tls_certs_cm {
   for app in "${apps[@]}"; do
     # Only create the secrets if they do not exist
     TLS_SECRET_NAME=$app-tls-secret
-    if [ -z "$(kubectl get secret -n $namespace $TLS_SECRET_NAME -o name 2>/dev/null)" ]; then
-        if [ "$deployedIssuers" == "false" ]; then
-          deploy_issuers $namespace $context
-          deployedIssuers="true"
-        fi
-        deploy_app_cert "$namespace" "$context" "$app"
-    else
-      log_debug "Using existing $TLS_SECRET_NAME for [$app]"
+    if [ -n "$(kubectl get secret -n $namespace $TLS_SECRET_NAME -o name 2>/dev/null)" ]; then
```

Comment:
I flipped the logic of this if block so that it would check if the secret does exist and then check if certs are expired or not.

Reply:
The only TLS certs we should replace are those that we generated.  So, if the organization has deployed using their own certs (i.e. not had us generate certs using openssl or cert-manager), we should NOT replace their certs even if they are expired or about to expire.   I believe (but please confirm) that the secrets containing the certs we generate (regardless of cert generation mechanism) have a label of "managed-by" with a value of "v4m" (i.e. "managed-by=v4m").    

In addition, if cert-manager is being used, cert-manager will actually generate new certs when the old certs expire.  The challenge is that these certs are not picked up by the pod automatically, so while we do not have to regenerate the certs, the pods need to be restarted to pick them up.  But I don't know if we really want to write up code to handle that...especially since we would (probably) have to deal with cases where the original certs were generated by cert-manager but the user has indicated that openssl should be used now.  

So, I think the take-aways here are: 

- do nothing if the certs were not generated by us, regardless of their expiration date.
- handle cert-manager and openssl generated certs the same way (and ignore the cert-manager functionality that renews certs for us).